# Vaihe 1

### Kiinnostavat tutkimuskysymykset
1. Mitkä genret ovat olleet suosittuja milloin?
2. Mikä näyttelijöiden yhdistelmä tuottaa suosittuja elokuvia?
3. Korrelaatio arvioiden ja tuoton välillä. Voiko elokuva olla huonosti arvioitu mutta silti tuottoisa tai toisin päin?
4. Jos tekisimme elokuvan, millainen sen kannattaa olla, että se on tuottoisa ja suosittu?

# Vaihe 2

### Aineiston kuvaus
Aineisto löydettiin Kagglesta ja se sisältää tietoa tuhannesta parhaiten arvioidusta elokuvasta ja TV -sarjasta IMDB palvelussa.

Linkki aineistoon: [LINKKI](https://www.kaggle.com/datasets/ashpalsingh1525/imdb-movies-dataset)


In [8]:
import numpy as np
import pandas as pd

dataset = pd.read_csv('Datasets/imdb_movies.csv')

# TODO: Fixaa tää silleen et selitetään otsikoiden tarkoitus
# Print dataframe columns
print("Aineiston otsikot:")
print(dataset.columns)
print(dataset.shape)


Aineiston otsikot:
Index(['names', 'date_x', 'score', 'genre', 'overview', 'crew', 'orig_title',
       'status', 'orig_lang', 'budget_x', 'revenue', 'country'],
      dtype='object')
(10178, 12)


Aineiston otsikot:

TODO: Lisää selitys otiskoista Crew Members otsikko on tosi outo, se pitää selittää jotenkin


Aineistossa on siis 12 otsikkoa ja 10178 havaintoa.



# Vaihe 3
## Aineiston esikäsittely

#### Tyhjien havaintojen poisto

In [10]:
# Tässä poistetaan ne havainnot joissa on tyhjiä arvoja tai arvostelu on 0.0

#### Avataan näyttelijät

In [9]:
# Tässä avataan näyttelijät

# Vaihe 4
### Aineiston analysointi

In [12]:
#  JOKAISESSA ANALYSOINTIKOHDASSA TEHDÄÄN AINEISTOSTA KOPIO, ETTEI MYÖHEMMÄT KOHDAT MUUTU

#### Mitkä genret ovat suosittuja milloin

In [11]:
# Tässä koodiblokki


#### Mikä näyttelijöiden yhdistelmä tuottaa suosittuja elokuvia?

#### Korrelaatio arvioiden ja tuoton välillä. Voiko elokuva olla huonosti arvioitu mutta silti tuottoisa tai toisin päin?

#### Jos tekisimme elokuvan, millainen sen kannattaa olla, että se on tuottoisa ja suosittu?